src/hera/workflows/script.py

```diff
@@ -232,16 +236,41 @@ def _build_script(self) -> _ModelScriptTemplate:
 def _get_parameters_from_callable(source: Callable) -> Optional[List[Parameter]]:
     # If there are any kwargs arguments associated with the function signature,
     # we store these as we can set them as default values for argo arguments
-    source_signature: Dict[str, Optional[object]] = {}
+    #
+    # Take all the annotation information and make sure it is stored.
+    parameters = []
+
     for p in inspect.signature(source).parameters.values():
         if p.default != inspect.Parameter.empty and p.kind == inspect.Parameter.POSITIONAL_OR_KEYWORD:
-            source_signature[p.name] = p.default
+            default = p.default
         else:
-            source_signature[p.name] = MISSING
+            default = MISSING
+
+        param = Parameter(name=p.name, default=default)
+        parameters.append(param)
+
+        if not global_config.experimental_features["script_annotations"]:
+            continue
+
+        if get_origin(p.annotation) is not Annotated or not isinstance(get_args(p.annotation)[1], Parameter):
+            continue
 
-    if len(source_signature) == 0:
+        annotation = get_args(p.annotation)[1]
+        for attr in ["enum", "description", "default", "name"]:
+            if not hasattr(annotation, attr) or getattr(annotation, attr) is None:
+                continue
+
+            if attr == "default" and param.default is not None:
+                raise ValueError(
+                    "The default cannot be set via both the function parameter default and the annotation's default"
+                )
+
+            setattr(param, attr, getattr(annotation, attr))
+
+    if len(parameters) == 0:
         return None
-    return [Parameter(name=n, default=v) for n, v in source_signature.items()]
+
+    return parameters
```

Comment:
Thanks to your changes we can now actually streamline this even more 😎

```suggestion
    return parameters or None
```